### ISSUE TITLE
as_list() omits highest-dimensional simplices

### ISSUE THREAD

**Issue by corybrunson** [peekxc/simplextree#12]
I'm building a workflow using the current development version and came across a bug with `$as_list()`: Only the `d - 1L`-dimensional simplices of a `d`-dimensional simplicial complex are returned (see the reprex below). If helpful, i'd be glad to submit a PR with a fix and a new test.

``` r
library(simplextree)
#> 
#> Attaching package: 'simplextree'
#> The following object is masked from 'package:utils':
#> 
#>     find
#> The following objects are masked from 'package:base':
#> 
#>     remove, serialize
st <- simplex_tree(list(1:3, 2:5, c(6, 7, 9), 7:8, 10))
st$n_simplices
#> [1] 10 12  6  1
st$as_list()
#> [[1]]
#>      [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10]
#> [1,]    1    2    3    4    5    6    7    8    9    10
#> 
#> [[2]]
#>      [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10] [,11] [,12]
#> [1,]    1    1    2    2    2    3    3    4    6     6     7     7
#> [2,]    2    3    3    4    5    4    5    5    7     9     8     9
#> 
#> [[3]]
#>      [,1] [,2] [,3] [,4] [,5] [,6]
#> [1,]    1    2    2    2    3    6
#> [2,]    2    3    3    4    4    7
#> [3,]    3    4    5    5    5    9
```

<sup>Created on 2020-11-06 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by peekxc** [comment]
This ended up being a corner case that happens due to only the way the `as_list` function works relative to the internal BFS traversal, written eons ago. 

I've removed it, since philosophically one has better control over simplex retrieval using the `as.list.st_traversal` S3 generic, which is also more uniform in that it always reports one simplex per entry in a list form. Indeed, though matrix formats are nice an compact, there is no standard on row vs column orientation, row being more familiar to data scientists and column more familiar to scientific computing and combinatorics people (and more efficient to report, since R is column major)

You can emulate that function now by rolling your own:
 
``` r
library(simplextree)
#> 
#> Attaching package: 'simplextree'
#> The following object is masked from 'package:utils':
#> 
#>     find
#> The following objects are masked from 'package:base':
#> 
#>     remove, serialize
st <- simplex_tree(list(1:3, 2:5, c(6, 7, 9), 7:8, 10))
lapply(seq(0L, st$dimension), function(d){
  do.call(cbind, as.list(k_simplices(st, d)))
})
#> [[1]]
#>      [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10]
#> [1,]    1    2    3    4    5    6    7    8    9    10
#> 
#> [[2]]
#>      [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8] [,9] [,10] [,11] [,12]
#> [1,]    1    1    2    2    2    3    3    4    6     6     7     7
#> [2,]    2    3    3    4    5    4    5    5    7     9     8     9
#> 
#> [[3]]
#>      [,1] [,2] [,3] [,4] [,5] [,6]
#> [1,]    1    2    2    2    3    6
#> [2,]    2    3    3    4    4    7
#> [3,]    3    4    5    5    5    9
#> 
#> [[4]]
#>      [,1]
#> [1,]    2
#> [2,]    3
#> [3,]    4
#> [4,]    5
```

<sup>Created on 2021-07-21 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

Replacing `cbind` w/ `rbind` as needed. Note the use of `as.list` method dispatch here.

The row-oriented outputs for low dimensional simplices like `st$edges` should still remain attached at the module level.

**Comment by corybrunson** [comment]
This makes sense! I'll close the issue once i've tried it out on my machine.

**Comment by corybrunson** [comment]
This has been fully resolved in #13 (implementation) and #16 (debugging).
